Add language choice for installation

CodeShakingSheep opened the pull request:
## Problem

Nextcloud shows the following warning:

Your installation has no default phone region set. This is required to validate phone numbers in the profile settings without a country code. To allow numbers without a country code, please add "default_phone_region" with the respective ISO 3166-1 code of the region to your config file. For more details see the [documentation ↗](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2#Officially_assigned_code_elements).

![image](https://github.com/YunoHost-Apps/nextcloud_ynh/assets/19874562/32684594-c90f-4735-a54f-a9c1e5c30ffb)

## Solution

Added question for language in `manifest.toml`. Set 3 language variables (see here: https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/config_sample_php_parameters.html#default-language ) in `install`:
- `default_language`
- `default_locale`
- `default_phone_region`

~~In `upgrade` I set `default_phone_region` if it's not set according to `default_locale` (which always has a value as it has a default value).~~

Edit: Setting `default_phone_region` for existing installations should be handled in config panel, cf. #638. 

## PR Status

- [ ] Code finished and ready to be reviewed/tested
- [ ] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

CodeShakingSheep commented:
!testme

CodeShakingSheep commented:
!testme

CodeShakingSheep commented:
!testme

ericgaspar commented on scripts/upgrade:
```diff
@@ -143,6 +143,10 @@ then
 	config_locale=$(exec_occ config:system:get default_locale)
+	if [ "$config_locale" == "" ]; then
+		# If it's not set in config use "US" as default
+		config_locale="US"
```
Why not FR by default?

jarod5001 replied:
Why not any other code. It would be more logical that the user chooses from a drop down list at the installation or change it in the app panel

ericgaspar replied:
We are talking about a default value set if there was no previous value set.

CodeShakingSheep replied:
Hi @ericgaspar,
I set "US" because in `manifest.toml` I used "en_US" as default install language and so I wanted to use the default country here as well.

ericgaspar replied:
let's then change the default language to `fr` (because it is by far the most used...)

CodeShakingSheep commented:
!testme

CodeShakingSheep commented:
> Alrighty! [![Test Badge](https://camo.githubusercontent.com/3d787811b541311649599c9d40c8346f76a32ff512b752e11d17559d5f444c1a/68747470733a2f2f696d672e736869656c64732e696f2f656e64706f696e743f75726c3d68747470733a2f2f63692d617070732d6465762e79756e6f686f73742e6f72672f63692f6170692f6a6f622f31333732372f6261646765)](https://ci-apps-dev.yunohost.org/ci/job/13727) [![](https://camo.githubusercontent.com/28f674c60918cbe673183f57189c2240a2283ce2d705233a97fb2865b035287f/68747470733a2f2f63692d617070732d6465762e79756e6f686f73742e6f72672f63692f73756d6d6172792f31333732372e706e67)](https://ci-apps-dev.yunohost.org/ci/job/13727)

What is going on on the test machine? There is this error which is completely unrelated to this package. Can someone look into that?

```
WARNING W: GPG error: https://packages.sury.org/php bullseye InRelease: The following signatures were invalid: EXPKEYSIG B188E2B695BD4743 DEB.SURY.ORG Automatic Signing Key <deb@sury.org>
```

Salamandar commented on scripts/install:
```diff
@@ -87,6 +87,17 @@ ynh_script_progression --message="Configuring $app..." --weight=8
 
+# Set default language in config.php
+raw_language=$(echo "$language" | awk -F'_' '{print $1}')
+exec_occ config:system:set default_language --value="$raw_language"
```
shouldn't it be just $language here ?

CodeShakingSheep replied:
Nope, Nextcloud's `default_language` uses ISO_639-1 language codes such as `en` for English, see https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/config_sample_php_parameters.html#default-language

That's why we need to cut the first part from `$language`, e.g. take `en` from `en_US`.

Tagadda commented:
I restarted the last CI job https://ci-apps-dev.yunohost.org/ci/job/13727

CodeShakingSheep commented:
> is this language choice not best handled with config panels? cf. #638

Thanks for the hint. I hadn't checked out that PR. Indeed it seems to be the cleaner solution to set a value in config panel rather than to use a fixed default value during upgrade. So, I removed the fix in `upgrade` and renamed this PR accordingly.

CodeShakingSheep commented:
!testme

alexAubin commented:
Closing because it's been almost one year and it should really be something in the config panel rather than in the install questions ... in fact it's handled in https://github.com/YunoHost-Apps/nextcloud_ynh/pull/638 which does a bunch of other things, so closing in favor or #638
